Update file format docs for Iceberg (#19173)
trinodb/trino: docs/src/main/sphinx/connector/iceberg.md

configuration, custom properties, and snapshots of the table contents.

Iceberg data files are stored in either Parquet, ORC, or Avro format, as

> determined by the `iceberg.file-format` property in the catalog properties file.

Author: Maybe we need to also say that the table property is format and the column for the metadata is file_format .. its a mess..

Reviewer: I think we should rather stick to the old version of this sentence.
We can have `icebergb.file-format`   set to PARQUET, but create explicitly a table with `format` AVRO.

Author: The old is not really correct either, but I am moving these details in later sections.

Table properties supply or set metadata for the underlying tables. This is key
for {doc}`/sql/create-table-as` statements. Table properties are passed to the

> connector using a {doc}`WITH </sql/create-table-as>` clause:

Reviewer: Unrelated: It feels a bit weird to have the _Table properties_ section after _ALTER TABLE SET PROPERTIES_ section which references some of the properties even though they haven been introduced.

Author: Agreed.. but not going to fix that problem here..

connector using a {doc}`WITH </sql/create-table-as>` clause.

```
CREATE TABLE tablename

> WITH (format='CSV',

Author: This is definitely wrong for Iceberg...

Reviewer: That's good removal.
We have anyway an example of concrete table properties after the _Iceberg table properties_ table.

  * - Property name
    - Description

>   * - ``file_format``

Author: I am guessing..

Reviewer 2: `format` was correct.

All Iceberg table properties are defined here: https://github.com/trinodb/trino/blob/b7e5ee55760e9c2b04214359bb21fd11f2c0f4c1/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableProperties.java#L41

    - Description
  * - ``format``
    - Optionally specifies the format of table data files; either ``PARQUET``,
      ``ORC`, or ``AVRO``. Defaults to ``PARQUET`` from the default value

>       of ``iceberg.file-format`` or uses its setting from the catalog properties file.

Author: Do we need to tie it to the file_format metadata?

Reviewer: > Do we need to tie it to the file_format metadata?

I'm assuming here that you are referring to `$files` metadata table's `file_format` property.
In this case, my answer would be _no_.

>       of ``iceberg.file-format`` or uses its setting from the catalog properties file.

Reviewer: > or uses its setting from the catalog properties file.

WDYM?

>       ``ORC`, or ``AVRO``. Defaults to ``PARQUET`` from the default value
>       of ``iceberg.file-format`` or uses its setting from the catalog properties file.

Reviewer: ```suggestion
``ORC`, or ``AVRO``. Defaults to the value of ``iceberg.file-format`` 
configuration property which is by default ``PARQUET``.
```

Author: Updated

      Defaults to ``0.05``.
```

> The table definition below specifies to use Parquet files, partitioning by columns

Reviewer: ```suggestion
The table definition below specifies to use Parquet data files, partitioning by columns
```

Author: We just say Parquet files and ORC files everywhere else .. should I update across the Iceberg connector docs?

Reviewer: > should I update across the Iceberg connector docs?

Probably not.
In the context above it should be obvious that we're speaking about data files.
Actually the end user shouldn't be concerned about the fact that there are as well metadata files.

    location = '/var/example_tables/test_table')
```

> The table definition below specifies to use ORC files, bloom filter index by columns

Reviewer: ```suggestion
The table definition below specifies to use ORC data files, bloom filter index by columns
```
